x-pack/solutions/security/plugins/security_solution/server/endpoint/services/actions/clients/microsoft/defender/endpoint/ms_defender_endpoint_actions_client.ts

```diff
@@ -587,6 +668,9 @@ export class MicrosoftDefenderEndpointActionsClient extends ResponseActionsClien
   ): Promise<
     ActionDetails<ResponseActionRunScriptOutputContent, ResponseActionRunScriptParameters>
   > {
+    const scriptName = (actionRequest.parameters as MSDefenderRunScriptActionRequestParams)
```

Comment:
I think you could avoid the casting here by getting the script name from `reqIndexOptions` below (since that type is already set correctly)